tapgarden/batch.go

```diff
@@ -104,7 +105,10 @@ func (m *MintingBatch) validateGroupAnchor(s *Seedling) error {
 
 // MintingOutputKey derives the output key that once mined, will commit to the
 // Taproot asset root, thereby creating the set of included assets.
-func (m *MintingBatch) MintingOutputKey() (*btcec.PublicKey, []byte, error) {
+func (m *MintingBatch) MintingOutputKey(
+	sibling *commitment.TapscriptPreimage) (*btcec.PublicKey, []byte,
```

Comment:
As-is it would be persisted on `LoadTapscriptTree` right after we call `Finalize`. I agree we'd want to rework that if we want to be able to supply the sibling preimage earlier than the `Finalize` call (new RPC / CLI cmd?).

Reply:
Leaning that way to make it explicitly part of the batch struct vs. associated with the internal key.